merchant-app/app/src/main/java/com/googleinterns/smb/adapter/OrderAdapter.java

```diff
@@ -0,0 +1,161 @@
+package com.googleinterns.smb.adapter;
+
+import android.annotation.SuppressLint;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.CheckBox;
+import android.widget.CompoundButton;
+import android.widget.ImageButton;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.FragmentManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.bumptech.glide.Glide;
+import com.googleinterns.smb.R;
+import com.googleinterns.smb.fragment.EditPriceDialogFragment;
+import com.googleinterns.smb.model.BillItem;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Recycler view adapter for displaying bill items in billing activity
+ */
+public class OrderAdapter extends RecyclerView.Adapter<OrderAdapter.ViewHolder> {
+
+    public interface PriceChangeListener {
+        void onPriceChange(double newTotalPrice);
+    }
+
+    // fragment manager required for displaying dialogs
+    private FragmentManager mFragmentManager;
+    private List<BillItem> billItems;
+    private List<Boolean> availableItems;
+    private PriceChangeListener mListener;
+
+    public OrderAdapter(PriceChangeListener listener, List<BillItem> billItems, FragmentManager fragmentManager) {
+        mFragmentManager = fragmentManager;
+        this.billItems = billItems;
+        availableItems = new ArrayList<>();
+        for (int i = 0; i < billItems.size(); i++) {
+            availableItems.add(true);
+        }
+        mListener = listener;
+        // notify listener with initial total price
+        mListener.onPriceChange(getTotalPrice());
+    }
+
+    /**
+     * Compute total price by summing price for each item
+     */
+    private Double getTotalPrice() {
+        Double totalPrice = 0.0;
+        for (int i = 0; i < billItems.size(); i++) {
+            if (availableItems.get(i)) {
+                totalPrice += billItems.get(i).getTotalPrice();
+            }
+        }
```

Comment:
This seems more to me like Stream (Filter, Map, Collect) Usecase

Reply:
I tried using that, but android studio complains about requirement of android API level >= 24 for using that.